components/sup/src/config.rs

```diff
@@ -120,5 +120,33 @@ impl fmt::Display for GossipListenAddr {
 #[derive(Default, Debug, PartialEq, Eq)]
 pub struct Config {
     // Currently only used by `ServiceConfig`
-    pub service_config_http_listen: http_gateway::ListenAddr,
+    pub service_config_http_listen: HttpGatewayListenAddr,
+    // Currently only used by `Package`
+    pub package_config_from: Option<String>,
+}
+
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct HttpGatewayListenAddr {
```

Comment:
This is called the [newtype pattern](https://aturon.github.io/features/types/newtype.html). In that pattern, it's recommended to use the tuple style struct instead of named fields